mozci/util/hgmo.py

```diff
@@ -12,14 +13,56 @@
 
 HgPush = NewType("HgPush", Dict[str, Any])
 
+RE_SOURCE_REPO = re.compile(r"^Source-Repo: (https?:\/\/.*)$", re.MULTILINE)
```

Comment:
Could you mention where the code comes from? So in the future if there are changes to the original code, we know that we can apply them here too